### DISCUSSION TITLE
[GitHub REST API] `author_association` returns inconsistent value

### DISCUSSION THREAD

**Discussion by tim-finnigan** [community/community#18690] [Projects and Issues]
Sometimes this value returns "None" even though the user is actually a "Member" (see: https://api.github.com/repos/boto/botocore/issues/comments/1156849020)

I've also seen a user listed as a "Member" but it returns the `author_association` as "Contributor"

Can anyone explain this behavior or point me in the right direction as far as what to do regarding this?

**Comment by aktech** [comment]
I also had this issue and realised that the value of `author_association` depends on user's personal preference on their **Organization visibility**.

For example, if in my organization settings if I have set the "**Organization visibility**" as private, then the `author_association` for me will be `NONE` for any comment I make on any repository in that organization.
<img width="990" alt="Screenshot 2022-11-24 at 10 36 18 pm" src="https://user-images.githubusercontent.com/5647941/203870848-6a260853-f67e-46d7-9ba8-bb2a27633c7f.png">

**Comment by davidberenstein1957** [comment]
I have a similar issue
